automata/symbol_embedding/vector_databases.py

```diff
@@ -13,7 +14,15 @@
 V = TypeVar("V", bound=SymbolEmbedding)
 
 
-class ChromaSymbolEmbeddingVectorDatabase(ChromaVectorDatabase[str, V]):
+class IEmbeddingLookupProvider(abc.ABC):
+    """A concrete base class an interface for embedding lookup providers."""
+
+    def embedding_to_key(self, entry: SymbolEmbedding) -> str:
+        """Concrete implementation to generate a simple hashable key from a Symbol."""
+        return entry.symbol.dotpath
+
+
+class ChromaSymbolEmbeddingVectorDatabase(ChromaVectorDatabase[str, V], IEmbeddingLookupProvider):
     """Concrete class to provide a vector database that saves into a Chroma db."""
 
     def __init__(
```

Comment:
btw, I think we should generate the collection_name based on the repository we're indexing.

Reply:
This is a good strategy, we can implement that downstream when we fill the db, but I think this will be a good approach.